pkg/client/client.go

```diff
@@ -229,21 +364,76 @@ func (c *SPDKClient) EngineWatch(ctx context.Context) (*api.EngineStream, error)
 	return api.NewEngineStream(stream), nil
 }
 
-func (c *SPDKClient) EngineReplicaDelete(engineName, replicaName, replicaAddress string) error {
+func (c *SPDKClient) EngineSnapshotCreate(name, snapshotName string) error {
+	if name == "" || snapshotName == "" {
+		return fmt.Errorf("failed to create SPDK engine snapshot: missing required parameter name or snapshot name")
+	}
+
 	client := c.getSPDKServiceClient()
 	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
 	defer cancel()
 
-	_, err := client.EngineReplicaDelete(ctx, &spdkrpc.EngineReplicaDeleteRequest{
+	_, err := client.EngineSnapshotCreate(ctx, &spdkrpc.SnapshotRequest{
+		Name:         name,
+		SnapshotName: snapshotName,
+	})
+	return errors.Wrapf(err, "failed to create SPDK engine %s snapshot %s", name, snapshotName)
+}
+
+func (c *SPDKClient) EngineSnapshotDelete(name, snapshotName string) error {
+	if name == "" || snapshotName == "" {
+		return fmt.Errorf("failed to delete SPDK engine snapshot: missing required parameter name or snapshot name")
+	}
+
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
+	defer cancel()
+
+	_, err := client.EngineSnapshotDelete(ctx, &spdkrpc.SnapshotRequest{
+		Name:         name,
+		SnapshotName: snapshotName,
+	})
+	return errors.Wrapf(err, "failed to delete SPDK engine %s snapshot %s", name, snapshotName)
+}
+
+func (c *SPDKClient) EngineReplicaAdd(engineName, replicaName, replicaAddress string) error {
+	if engineName == "" {
+		return fmt.Errorf("failed to add replica for SPDK engine: missing required parameter engine name")
+	}
+	if replicaName == "" || replicaAddress == "" {
+		return fmt.Errorf("failed to add replica for SPDK engine: missing required parameter replica name or address")
+	}
+
+	client := c.getSPDKServiceClient()
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceTimeout)
+	defer cancel()
+
+	_, err := client.EngineReplicaAdd(ctx, &spdkrpc.EngineReplicaAddRequest{
 		EngineName:     engineName,
 		ReplicaName:    replicaName,
 		ReplicaAddress: replicaAddress,
 	})
-	if err != nil {
-		return errors.Wrap(err, "failed to delete replica from engine")
+	return errors.Wrapf(err, "failed to add replica %s with address %s to engine %s", replicaName, replicaAddress, engineName)
+}
+
+func (c *SPDKClient) EngineReplicaDelete(engineName, replicaName, replicaAddress string) error {
+	if engineName == "" {
+		return fmt.Errorf("failed to delete replica from SPDK engine: missing required parameter engine name")
 	}
+	if replicaName == "" || replicaAddress == "" {
```

Comment:
Should be `if replicaName == "" && replicaAddress == "" {`
Either of the two parameters is enough.

Reply:
Hello @shuo-wu 
Forgot update it.